### PR TITLE
Merge `main` back into `develop` post-1.5.1

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #DNE 
# What's Changed
- Keep `develop` in back in sync with `main`.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by gsnider2195 — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "nautobot"
 # Primary package version gets set here. This is used for publishing, and once
 # installed, `nautobot.__version__` will have this version number.
-version = "1.5.1-beta.1"
+version = "1.5.1"
```
Comment (bryanculver) [inline pyproject.toml:5]:
```suggestion
version = "1.5.2-beta.1"
```

**Review by bryanculver — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (glennmatthews) [reply pyproject.toml:5]: We don't want to make this change in `main` though!

[the remark this answers is not in the document]
Comment (jathanism) [reply pyproject.toml:5]: Would this be better to branch from `main`, or just do it in a follow-up PR in `develop` to bump the version?

[the remark this answers is not in the document]
Comment (glennmatthews) [reply pyproject.toml:5]: Either is fine by me.

**Pull request closed without merging**